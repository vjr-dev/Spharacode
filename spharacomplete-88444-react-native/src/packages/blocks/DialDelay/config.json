{
  "id": "DialDelay",
  "verion": "0.0.1",
  "status": "preAlpha",
  "path": "./packages/blocks/DialDelay",
  "assembler-adapters": [
    "DialDelayAdapter"
  ],
  "mobile": {
    "merge-engine-navigation": {},
    "assembler-exports": [
      "DialDelay"
    ],
    "assembler-dependencies": [],
    "catalog-dependencies": [
      "core",
      "DialDelay"
    ],
    "assembler-devDependencies": [],
    "merge-engine-screens": {
      "DialDelay": {
        "file": "DialDelay.tsx",
        "extends": "DialDelayController",
        "configURL": "",
        "validationRulesURL": "",
        "imports": [
          {
            "value": "DialDelayController",
            "path": "./DialDelayController"
          },
          {
            "value": "{Props}",
            "path": "./DialDelayController"
          }
        ]
      }
    }
  }
  
}
