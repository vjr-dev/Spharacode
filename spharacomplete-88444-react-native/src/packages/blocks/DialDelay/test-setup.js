// test-setup.js
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NativeModules } from 'react-native';

configure({ adapter: new Adapter() });



NativeModules.RNCAsyncStorage = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    removeItem: jest.fn(),
    mergeItem: jest.fn(),
    clear: jest.fn(),
    getAllKeys: jest.fn(),
    flushGetRequests: jest.fn(),
    multiGet: jest.fn(),
    multiSet: jest.fn(),
    multiRemove: jest.fn(),
    multiMerge: jest.fn(),
};



jest.mock('@react-native-community/slider', () => {
    const MockedSlider = () => <></>;
    return MockedSlider;
  });
