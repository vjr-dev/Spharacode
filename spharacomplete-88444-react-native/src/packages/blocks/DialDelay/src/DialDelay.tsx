// Customizable Area Start
//@ts-nocheck
import React from "react";

import {
    StyleSheet,
    Text,
    View,
    Image,
    ImageBackground,
    Dimensions,
    SafeAreaView,
    TouchableOpacity,
    Modal,
    ActivityIndicator,
    StatusBar
    
} from "react-native";

import DialDelayController, { Props } from "./DialDelayController";
import { back1, back2, image_back } from "./assets";
import { COLORS } from "../../../framework/src/Globals";
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;
import { RFValue } from "react-native-responsive-fontsize";
import Slider from '@react-native-community/slider';



export default class DialDelay extends DialDelayController {
    constructor(props: Props) {
        super(props);
    }

    

    render() 
    
    
    {



    let left: any
    if(this.state.value1 * RFValue(1.5) - RFValue(80) > RFValue(210))
    {
    left = RFValue(210)
    }
    else 
    {
        if(this.state.value1 * RFValue(1.5) - RFValue(80) > RFValue(1))
        {
            left = this.state.value1 * RFValue(1.5) - RFValue(80)
           }
           else
           {
            left = RFValue(1)
          }
    }

    let txtValue: any
    if (this.state.value1 <= 59) {
      txtValue = this.state.value1 + " second";
    } else if (this.state.value1 <= 119) {
      txtValue = "1 minute";
    } else if (this.state.value1 == 180) {
      txtValue = "3 minutes";
    } else {
      txtValue = "2 minutes";
    }
    

        return (
            <SafeAreaView style={styles.container1}>
                <View style={styles.child}>
                    <StatusBar
                        animated={true}
                        backgroundColor="#454545"
                        barStyle={"light-content"}
                        showHideTransition={"slide"}
                    
                    />
                    <ImageBackground
                        source={back1}
                        style={styles.image}>
                        <ImageBackground
                            source={back2}
                            style={styles.image2}>

                            <SafeAreaView>
                                <View
                                    style={styles.view1}
                                >
                                    <TouchableOpacity style={styles.button1} onPress={() => this.goback()} testID="bckbtnID">
                                        <Image
                                            source={image_back}
                                            style={styles.backbutton}
                                        />
                                    </TouchableOpacity>
                                    <TouchableOpacity
                                        style={{ paddingHorizontal: 25, justifyContent: "center" }}
                                        testID="btnSetID"
                                        onPress={() => this.setClick()}>
                                        <Text style={{ color: COLORS.darkorange, fontSize: 18 }}>SET</Text>
                                    </TouchableOpacity>
                                </View>

                                <Text style={{ fontSize: 20, color: COLORS.white, marginTop: RFValue(50), width: windowWidth - 20, alignSelf: "center" }}>Dial Delay.</Text>
                                <Text style={{ fontSize: 16, color: COLORS.ultralightwhite, marginTop: RFValue(10), width: windowWidth - 20, alignSelf: "center" }}>This setting will enable number of seconds delay need to connect with your local emergency number.</Text>

                                <View style={{
                                    height: RFValue(100), width: windowWidth,
                                    backgroundColor: COLORS.backgroundGray,
                                    
                                    marginTop: RFValue(60)
                                }}>
                                    <View style={{ height: "100%", width: windowWidth - RFValue(50), alignSelf: 'center', alignItems: "center", justifyContent: "space-evenly", flexDirection: "row" }}>
                                        <Text style={{ color: COLORS.lightwhite, marginLeft: RFValue(15), marginTop: RFValue(16) }}>30 sec.</Text>
                                        <View style={{ height: "100%", width: RFValue(300), justifyContent: "center" }}>
                                            <View style={{ width: RFValue(100), left: left, backgroundColor: COLORS.orangelight, borderRadius: RFValue(50) }}>
                                                <Text style={{ textAlign: 'center', color: COLORS.black, }}>{txtValue}</Text>
                                            </View>
                                            <View style={{ width: RFValue(300), alignItems: "center", marginTop: 5 }}>
                                                <Slider
                                                    style={{ width: RFValue(235), alignItems: "center", justifyContent: "center" }}
                                                    value={this.state.value1}
                                                    minimumValue={30}
                                                    maximumValue={180}
                                                    onValueChange={(V) => this.setState({ value1: V })}
                                                    minimumTrackTintColor={COLORS.darkorange}
                                                    maximumTrackTintColor={COLORS.ultralightwhite}
                                                    thumbTintColor={COLORS.white}

                                                    step={1}
                                                /></View>
                                        </View>
                                        <Text style={{ color: COLORS.lightwhite, marginRight: RFValue(15), marginTop: RFValue(16) }}>3 min.</Text>
                                    </View>
                                </View>

                                <Modal visible={this.state.Loader} transparent={true}>
                                    <View style={{ height: windowHeight, width: windowWidth, alignItems: "center", justifyContent: "center" }}>
                                        <View
                                            style={{ height: 100, width: windowWidth - 20, alignSelf: "center", justifyContent: "center", alignItems: "center", }} >
                                            <ActivityIndicator
                                                animating={true}
                                                size={"large"}
                                                color="#f07233"
                                            />
                                        </View>
                                    </View>
                                </Modal>
                            </SafeAreaView>

                        </ImageBackground>
                    </ImageBackground >
                </View >
            </SafeAreaView >
            
        );
    }
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: windowHeight, width: windowWidth,
        backgroundColor: "#ffffffff"
    },
    image: {
        height: windowHeight, width: windowWidth,
    },
    image2: {
        height: windowHeight, width: windowWidth, alignItems: "center",
    },
    container1: { flex: 1, backgroundColor: "#454545" },
    child: { width: windowWidth, },
    backbutton: { height: 15, width: 15, tintColor: "#f07233" },
    view1: { height: 60, width: windowWidth, justifyContent: "space-between", flexDirection: "row" },
    button1: {
        width: 15,
        height: 15,
        padding: 25,
        justifyContent: "center",
        alignItems: "center"
    },

});
// Customizable Area End
