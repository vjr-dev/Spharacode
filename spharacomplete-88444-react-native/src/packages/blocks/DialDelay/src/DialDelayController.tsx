// Customizable Area Start
//@ts-ignore
//@ts-nocheck
import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
    getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";


import AsyncStorage from '@react-native-async-storage/async-storage'
import { Alert } from "react-native";



export const configJSON = require("./config");


export interface Props {
    navigation: any;
    id: string;
    
    
}

interface S {
    
    Loader: boolean
    value1: number
    Token: any
    Userdata: any

    
}

interface SS {
    id: any;
    
    
}

export default class DialDelay extends BlockComponent<Props, S, SS> {
    
    SetsettingID: any = "";
    

    constructor(props: Props) {
        super(props);
        this.receive = this.receive.bind(this);

        
        this.subScribedMessages = [
            
            getName(MessageEnum.RestAPIResponceMessage),
            getName(MessageEnum.RestAPIRequestMessage),
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            getName(MessageEnum.RestAPIRequestBodyMessage),
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            getName(MessageEnum.RestAPIRequestMethodMessage),
            
        ];

        this.state = {
            
            Loader: false,
            value1: 30,
            Userdata: "",
            Token: "",

            
        };
        runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);

        
        
    }


    


     componentDidMount() {
         this.setState({ Userdata: this.props.route.params.data })
        
        this.setState({ Token: AsyncStorage.getItem("Token") })
        this.setState({ value1: this.state.Userdata.dial_delay })
    }
    //istanbul ignore next
    async receive(from: string, message: Message) {
        
        runEngine.debugLog("Message Recived", message);

        if (message.id === getName(MessageEnum.RestAPIResponceMessage)) {
            const apiRequestCallId = message.getData(
                getName(MessageEnum.RestAPIResponceDataMessage)
            );

            let responseJson = message.getData(
                getName(MessageEnum.RestAPIResponceSuccessMessage)
            );

            let errorReponse = message.getData(
                getName(MessageEnum.RestAPIResponceErrorMessage)
            );

            if (apiRequestCallId === this.SetsettingID) {
                if (responseJson != null) {

                    if (responseJson.data != null) {

                        this.setState({ Loader: false })
                    } else {
                        this.setState({ Loader: false })
                        Alert.alert(errorReponse)
                    }


                }
            }
        }
    }

    async setClick() {

        this.setState({ Loader: true })
        const header = {
            "Content-Type": configJSON.UpdatesettingAPiContentType,
            token: this.state.Token
        };
        const attrs = {
            dial_delay: this.state.value1,
        };
        const data1 = {
            attributes: attrs
        };
        const httpBody = {
            data: data1
        };
        console.log(httpBody, "httpBody");

        const requestMessage = new Message(
            getName(MessageEnum.RestAPIRequestMessage)
        );

        this.SetsettingID = requestMessage.messageId;


        requestMessage.addData(
            getName(MessageEnum.RestAPIResponceEndPointMessage),
            configJSON.UpdatesettingAPiEndPoint
        );
        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestHeaderMessage),
            JSON.stringify(header)
        );

        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestBodyMessage),
            JSON.stringify(httpBody)
        );

        requestMessage.addData(
            getName(MessageEnum.RestAPIRequestMethodMessage),
            configJSON.UpdatesettingAPiMethod
        );

        runEngine.sendMessage(requestMessage.id, requestMessage);

        return true;

    }


    goback() {
        const { navigation }: any = this.props
        navigation.goBack()
    }
    
}
// Customizable Area End