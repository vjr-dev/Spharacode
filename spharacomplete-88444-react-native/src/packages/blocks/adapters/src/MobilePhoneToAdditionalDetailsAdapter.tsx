import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class MobilePhoneToAdditionalDetailsAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationMobilePhoneAdditionalDetailsMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'AdditionalDetailForm'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );
    raiseMessage.addData(
      getName(MessageEnum.AuthTokenDataMessage),
      from.getData(getName(MessageEnum.AuthTokenDataMessage))
    );

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
