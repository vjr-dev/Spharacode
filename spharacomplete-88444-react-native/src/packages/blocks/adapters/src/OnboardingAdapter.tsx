import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class OnboardingAdapter {
  send: (message: Message) => void;

  constructor() {
    // const uuidv4 = require('uuid/v4');
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.OnboardingGuideControllerDoneOnboarding),
      getName(MessageEnum.OnboardingGuideControllerSkipOnboarding)
    ]);
  }

  convert = (from: Message): Message => {
    runEngine.debugLog("OnboardingAdapter", from.id)
    const to = new Message(getName(MessageEnum.NavigationMessage));
    to.addData(getName(MessageEnum.NavigationTargetMessage), 'Home');
    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );
    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
