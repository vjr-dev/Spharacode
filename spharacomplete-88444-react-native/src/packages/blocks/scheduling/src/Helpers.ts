/* App/Lib/GeneralHelpers.js */
import { Platform, StyleSheet } from 'react-native';
import { heightFromPercentage } from '../../../framework/src/Utilities';

// Colors Constants
export const COLORS = {
  TRANSPARENT: 'transparent',
  white: 'rgb(255,255,255)',
  black: 'rgb(0,0,0)',
  black10: 'rgb(10,10,10)',
  green: 'rgb(76,217,100)',
  red: 'red',
  yellow: 'rgba(255, 212, 40, 1)',
  darkGray: 'rgba(105,105,105,0.9)',
  lightGray: '#a7b6b9',
  skipGray: 'rgb(91,91,92)',
  borderColor: 'rgb(240, 240, 240)',
  inactiveIndicator: 'rgb(228, 233, 242)',
  inputLabel: 'rgb(143,155,179)',
  inputIcon: 'rgb(197,206,224)',
  inputValue: 'rgb(46,58,89)',
  discountRed: 'rgb(199,5,28)',
  orange: 'rgb(255,157,43)',
  infoGray: 'rgb(163,164,165)',
  ReviewBlack: 'rgb(34,43,69)',
  blue: 'blue',
  blueGrey: 'rgb(143,155,179)',
  darkBlueGrey: 'rgb(46,58,89)',
  lightBlueGrey: 'rgb(197, 206, 224)'
};

export const CommonStyle = StyleSheet.create({
  container: {
    flex: 1,
    width: "100%",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  wrapper: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  headerContainer:{
    backgroundColor:COLORS.yellow,
    height: '10%',
  },
  authHeader:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.yellow
  },
  authHeaderWhite:{
    height: heightFromPercentage(10),
    backgroundColor: COLORS.white
  },
  formContainer: {
    width: "80%"
  },
  textAlignCenter:{
    textAlign: 'center'
  },
  labelTextStyle: {
    color: COLORS.inputLabel
  },
  inputContainerStyle:{
    borderBottomWidth: 1,
    borderBottomColor: COLORS.inactiveIndicator
  },
  error: {
    color: COLORS.red,
    fontSize: 14
  },
  scrollStyle: {
    width: "100%"
  },
  btnTitleStyle:{
    fontSize: 17,
    color: COLORS.black
  },
  closeButtonStyle: {
    backgroundColor: COLORS.inputIcon,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  yellowButtonStyle: {
    backgroundColor: COLORS.yellow,
    height: 40,
    width: 100,
    borderRadius: 8
  },
  closeBtnTitleStyle:{
    color: COLORS.black
  },
  scrollContainer: {
    flexGrow: 1,
    alignItems: "center"
  },
  shadowStyle: {
    shadowOffset:
      Platform.OS === "ios"
        ? { width: 0, height: 1 }
        : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 3 : 5,
    elevation: 5
  },
  boxShadowStyle: {
    backgroundColor: COLORS.white,
    marginHorizontal:2,
    marginVertical:1,
    shadowOffset:
    Platform.OS === "ios"
      ? { width: 0, height: 1 }
      : { width: 0, height: 2 },
    shadowColor: COLORS.inputLabel,
    shadowOpacity: 0.9,
    shadowRadius: Platform.OS === "ios" ? 1 : 5,
    elevation: 2,    
    borderRadius:8,
  },
  toastModal: {
    borderRadius: 20,
    marginHorizontal: 15,
    paddingHorizontal: '12%',
    paddingVertical: 100,
    alignContent: "center",
    justifyContent: "center",
  },
  clearModal: {
    borderRadius: 5,
    marginHorizontal: 50,
    height: 220,
    backgroundColor: COLORS.red
  },
  imageThumbContain: {
    resizeMode: "contain",
    width: "100%",
    height: "100%"
  },
  imageThumbCover: {
    resizeMode: "cover",
    width: "100%",
    height: "100%"
  },
  flexContainer: {
    flex: 1,
    width: "100%"
  },
  rowStyle: {
    flexDirection: "row"
  },
  flexStart: {
    justifyContent: "flex-start"
  },
  flexEnd: {
    justifyContent: "flex-end"
  },
  jCCenter: {
    justifyContent: "center"
  },
  spaceBetween: {
    justifyContent: "space-between",    
  },
  spaceAround: {
    justifyContent: 'space-around'
  },
  alignItemsCenter:{
    alignItems: "center"
  },
  styleCenter: {
    justifyContent: "flex-end",
    alignItems: "center"
  },
  centerStyle: {
    justifyContent: "center",
    alignItems: "center"
  },
  width100P:{
    width: '100%'
  },
  subView: {
    flex: 1,
    width: "100%",
    height: "100%"
  },
  emptyList: {
    alignItems: "center",
    marginTop: 100
  },
  emptyListFilter: {
    alignItems: "center",
    justifyContent: "center",
    marginTop: 100
  },
  backIconView: {
    marginVertical: 10,
    marginHorizontal: 5,
    width: 30,
    height: 30,
    justifyContent: "flex-end",
    alignItems: "flex-end"
  },
  crossImage: {
    tintColor: COLORS.red,
    width: "100%",
    height: "100%",
    resizeMode: "contain"
  },
  searchTextInput: {
    width: "80%"
  },
  errorReport: {
    width: "70%",
    color: COLORS.red,
    fontSize: 15,
    marginLeft: "28.8%"
  }
});