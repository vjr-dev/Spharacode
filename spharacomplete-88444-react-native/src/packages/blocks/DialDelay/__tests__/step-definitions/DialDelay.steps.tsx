//@ts-ignore
//@ts-nocheck
import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'

import * as helpers from '../../../../framework/src/Helpers'
import React from "react";
import DialDelay from "../../src/DialDelay"
const navigation = require("react-navigation")


const screenProps = {
    navigation: {
        navigate: jest.fn(),
        goBack: jest.fn(),
        addListener: jest.fn().mockImplementation((event: any, callback: any) => {
            callback();
          }),
    },
    route: {
        params: {
          data: { name: 'Test Name' },
          userName: 'abc'
        }
    },
    id: "DialDelay"
  }

const feature = loadFeature('./__tests__/features/DialDelay-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });
    test('User navigates to DialDelay', ({ given, when, then }) => {
        let DialDelayBlockWrapper:ShallowWrapper;
        let instance:DialDelay ; 

        given('I am a User loading DialDelay', () => {
            //@ts-ignore
            DialDelayBlockWrapper = shallow(<DialDelay {...screenProps}/>)
            expect(DialDelayBlockWrapper).toBeTruthy()
            expect(DialDelayBlockWrapper).toMatchSnapshot()     
        });

        when('I navigate to the DialDelay', () => {
             instance = DialDelayBlockWrapper.instance() as DialDelay
      
        });

        then('DialDelay will load with out errors', () => {
            //@ts-ignore
            expect(DialDelayBlockWrapper).toBeTruthy()
          
        });

        then('Handle back button press',() => {
            let buttonComponent = DialDelayBlockWrapper.findWhere((node) => node.prop('testID') === 'bckbtnID');
            buttonComponent.simulate('press', instance.goback());
            //@ts-ignore
            expect(buttonComponent).toBeTruthy();
        })
        then('Handle the set button press', () => {
            instance.setState({value1: 300})
            let buttonComponent = DialDelayBlockWrapper.findWhere((node) => node.prop('testID') === 'btnSetID');
            buttonComponent.simulate('press');
            instance.setState({value1: 100})
            
            //@ts-ignore
            expect(buttonComponent).toBeTruthy();
            expect(DialDelayBlockWrapper).toMatchSnapshot()     
        })
    })
}
)
