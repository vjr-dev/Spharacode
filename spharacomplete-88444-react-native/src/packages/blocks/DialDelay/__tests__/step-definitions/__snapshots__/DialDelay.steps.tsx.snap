// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DialDelay User navigates to DialDelay 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/DialDelay/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/DialDelay/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <SafeAreaView>
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "width": 750,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 15,
                  "justifyContent": "center",
                  "padding": 25,
                  "width": 15,
                }
              }
              testID="bckbtnID"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/DialDelay/assets/image_back.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "tintColor": "#f07233",
                    "width": 15,
                  }
                }
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                  "paddingHorizontal": 25,
                }
              }
              testID="btnSetID"
            >
              <Text
                style={
                  Object {
                    "color": "rgba(255,138,73,1)",
                    "fontSize": 18,
                  }
                }
              >
                SET
              </Text>
            </TouchableOpacity>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "rgb(255,255,255)",
                "fontSize": 20,
                "marginTop": 98,
                "width": 730,
              }
            }
          >
            Dial Delay.
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#a1a1a1",
                "fontSize": 16,
                "marginTop": 20,
                "width": 730,
              }
            }
          >
            This setting will enable number of seconds delay need to connect with your local emergency number.
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#404040",
                "height": 196,
                "marginTop": 118,
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "row",
                  "height": "100%",
                  "justifyContent": "space-evenly",
                  "width": 652,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "marginLeft": 29,
                    "marginTop": 31,
                  }
                }
              >
                30 sec.
              </Text>
              <View
                style={
                  Object {
                    "height": "100%",
                    "justifyContent": "center",
                    "width": 589,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(255,211,154,1)",
                      "borderRadius": 98,
                      "left": 2,
                      "width": 196,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(0,0,0)",
                        "textAlign": "center",
                      }
                    }
                  >
                    2 minutes
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "marginTop": 5,
                      "width": 589,
                    }
                  }
                >
                  <MockedSlider
                    maximumTrackTintColor="#a1a1a1"
                    maximumValue={180}
                    minimumTrackTintColor="rgba(255,138,73,1)"
                    minimumValue={30}
                    onValueChange={[Function]}
                    step={1}
                    style={
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "width": 461,
                      }
                    }
                    thumbTintColor="rgb(255,255,255)"
                  />
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "marginRight": 29,
                    "marginTop": 31,
                  }
                }
              >
                3 min.
              </Text>
            </View>
          </View>
          <Component
            hardwareAccelerated={false}
            transparent={true}
            visible={false}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 1334,
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 100,
                    "justifyContent": "center",
                    "width": 730,
                  }
                }
              >
                <ActivityIndicator
                  animating={true}
                  color="#f07233"
                  hidesWhenStopped={true}
                  size="large"
                />
              </View>
            </View>
          </Component>
        </SafeAreaView>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;

exports[`DialDelay User navigates to DialDelay 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#454545",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "width": 750,
      }
    }
  >
    <StatusBar
      animated={true}
      backgroundColor="#454545"
      barStyle="light-content"
      showHideTransition="slide"
    />
    <ImageBackground
      source={
        Object {
          "testUri": "../../../packages/blocks/DialDelay/assets/image_background-1.png",
        }
      }
      style={
        Object {
          "height": 1334,
          "width": 750,
        }
      }
    >
      <ImageBackground
        source={
          Object {
            "testUri": "../../../packages/blocks/DialDelay/assets/image_background-2.png",
          }
        }
        style={
          Object {
            "alignItems": "center",
            "height": 1334,
            "width": 750,
          }
        }
      >
        <SafeAreaView>
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 60,
                "justifyContent": "space-between",
                "width": 750,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "height": 15,
                  "justifyContent": "center",
                  "padding": 25,
                  "width": 15,
                }
              }
              testID="bckbtnID"
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../packages/blocks/DialDelay/assets/image_back.png",
                  }
                }
                style={
                  Object {
                    "height": 15,
                    "tintColor": "#f07233",
                    "width": 15,
                  }
                }
              />
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Object {
                  "justifyContent": "center",
                  "paddingHorizontal": 25,
                }
              }
              testID="btnSetID"
            >
              <Text
                style={
                  Object {
                    "color": "rgba(255,138,73,1)",
                    "fontSize": 18,
                  }
                }
              >
                SET
              </Text>
            </TouchableOpacity>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "rgb(255,255,255)",
                "fontSize": 20,
                "marginTop": 98,
                "width": 730,
              }
            }
          >
            Dial Delay.
          </Text>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#a1a1a1",
                "fontSize": 16,
                "marginTop": 20,
                "width": 730,
              }
            }
          >
            This setting will enable number of seconds delay need to connect with your local emergency number.
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#404040",
                "height": 196,
                "marginTop": 118,
                "width": 750,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "flexDirection": "row",
                  "height": "100%",
                  "justifyContent": "space-evenly",
                  "width": 652,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "marginLeft": 29,
                    "marginTop": 31,
                  }
                }
              >
                30 sec.
              </Text>
              <View
                style={
                  Object {
                    "height": "100%",
                    "justifyContent": "center",
                    "width": 589,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "rgba(255,211,154,1)",
                      "borderRadius": 98,
                      "left": 143,
                      "width": 196,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "rgb(0,0,0)",
                        "textAlign": "center",
                      }
                    }
                  >
                    1 minute
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "marginTop": 5,
                      "width": 589,
                    }
                  }
                >
                  <MockedSlider
                    maximumTrackTintColor="#a1a1a1"
                    maximumValue={180}
                    minimumTrackTintColor="rgba(255,138,73,1)"
                    minimumValue={30}
                    onValueChange={[Function]}
                    step={1}
                    style={
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "width": 461,
                      }
                    }
                    thumbTintColor="rgb(255,255,255)"
                    value={100}
                  />
                </View>
              </View>
              <Text
                style={
                  Object {
                    "color": "#cdcdcd",
                    "marginRight": 29,
                    "marginTop": 31,
                  }
                }
              >
                3 min.
              </Text>
            </View>
          </View>
          <Component
            hardwareAccelerated={false}
            transparent={true}
            visible={true}
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 1334,
                  "justifyContent": "center",
                  "width": 750,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "height": 100,
                    "justifyContent": "center",
                    "width": 730,
                  }
                }
              >
                <ActivityIndicator
                  animating={true}
                  color="#f07233"
                  hidesWhenStopped={true}
                  size="large"
                />
              </View>
            </View>
          </Component>
        </SafeAreaView>
      </ImageBackground>
    </ImageBackground>
  </View>
</SafeAreaView>
`;
